src/pyrenew/process/ar.py

```diff
@@ -4,97 +4,92 @@
 from __future__ import annotations
 
 import jax.numpy as jnp
-import numpyro
-import numpyro.distributions as dist
-from jax import lax
 from jax.typing import ArrayLike
+from numpyro.contrib.control_flow import scan
 
 from pyrenew.metaclass import RandomVariable, SampledValue
+from pyrenew.process.iidrandomsequence import StandardNormalSequence
 
 
 class ARProcess(RandomVariable):
     """
-    Object to represent
-    an AR(p) process in
-    Numpyro
+    RandomVariable representing an
+    an AR(p) process.
     """
 
-    def __init__(
-        self,
-        name: str,
-        mean: float,
-        autoreg: ArrayLike,
-        noise_sd: float,
-    ) -> None:
+    def __init__(self, name, *args, noise_rv_suffix: str = "_noise", **kwargs):
         """
-        Default constructor
+        Default constructor.
 
         Parameters
         ----------
         name : str
-            Name of the parameter passed to numpyro.sample.
-        mean: float
-            Mean parameter.
-        autoreg : ArrayLike
-            Model parameters. The shape determines the order.
-        noise_sd : float
-            Standard error for the noise component.
+           A name for the process.
 
-        Returns
-        -------
-        None
+        noise_rv_suffix : str
+           A suffix to append to name when naming the
+           internal RandomVariable holding the process
+           noise. Default `"_noise"`.
         """
         self.name = name
-        self.mean = mean
-        self.autoreg = autoreg
-        self.noise_sd = noise_sd
+        super().__init__(*args, **kwargs)
+        self.noise_rv_ = StandardNormalSequence(name=name + noise_rv_suffix)
 
     def sample(
         self,
-        duration: int,
-        inits: ArrayLike = None,
+        n: int,
+        autoreg: ArrayLike,
+        init_vals: ArrayLike,
+        noise_sd: float | ArrayLike,
         **kwargs,
     ) -> tuple:
         """
         Sample from the AR process
 
         Parameters
         ----------
-        duration: int
+        n: int
             Length of the sequence.
-        inits : ArrayLike, optional
-            Initial points, if None, then these are sampled.
-            Defaults to None.
+        autoreg: ArrayLike
```

Comment:
Seems to be written as though it will work with a multidimensional `autoreg` and `init_vals`. If this is the case, we should have a test for it. If not, add more checks regarding the dimensionality.

Reply:
I think it can be made to work, but as written probably would not work "as expected". So for now I've made things stricter about 1D arrays and added tests that the errors get raised (in e0273ae)